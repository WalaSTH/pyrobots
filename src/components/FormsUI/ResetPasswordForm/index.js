import React from "react";
import Grid from "@mui/material/Grid";
import Card from "@mui/material/Card";
import Typography from "@mui/material/Typography";
import * as Yup from "yup";
import { Formik, Form } from "formik";
import TextField from "../TextField";
import SubmitFormButton from "../SubmitFormButton";

const initialFormState = {
  password: "",
  passwordConfirmation: "",
};

const formValidation = Yup.object().shape({
  password: Yup.string()
    .min(8, "Password must be at least 8 characters")
    .matches(
      /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})/,
      "Must Contain 8 Characters, one Uppercase, one Lowercase and one Number"
    ),
  passwordConfirmation: Yup.string().oneOf(
    [Yup.ref("password"), null],
    "Passwords must match"
  ),
});

export default function ResetPasswordForm({ handleSubmit }) {
  return (
    <Formik
      initialValues={{ ...initialFormState }}
      validationSchema={formValidation}
      onSubmit={handleSubmit}
    >
      <Card
        variant="outlined"
        sx={{
          marginTop: 3,
          padding: 3,
          borderRadius: 3,
          textAlign: "center",
        }}
      >
        <Form>
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <Typography
                component="h1"
                variant="h5"
                children="Reset your password"
              />
            </Grid>

            <Grid item xs={12}>
              <TextField
                name="password"
                data-testid="Password"
                label="Password"
                type="password"
                autoComplete="off"
                required
              />
            </Grid>

            <Grid item xs={12}>
              <TextField
                name="passwordConfirmation"
                data-testid="passwordConfirmation"
                label="Confirm your password"
                type="password"
                autoComplete="off"
                required
              />
            </Grid>

            <Grid item xs={12}>
              <SubmitFormButton data-testid="resetPassword-button">
                Reset Password
              </SubmitFormButton>
            </Grid>
          </Grid>
        </Form>
      </Card>
    </Formik>
  );
}
