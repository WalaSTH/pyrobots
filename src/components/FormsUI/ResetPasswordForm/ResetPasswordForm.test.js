import React from "react";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import ResetPasswordForm from "./";

describe("<ResetPassword", () => {
  const user = userEvent.setup();
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("component rendering", () => {
    const component = renderer.create(<ResetPasswordForm />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("component renders with all the components", () => {
    render(<ResetPasswordForm />);
    screen.getByText("Reset your password");
    screen.getByTestId("Password");
    screen.getByTestId("passwordConfirmation");
    screen.getByTestId("resetPassword-button");
  });

  it("validates password field to be not empty after write", async () => {
    const mockHandler = jest.fn();
    render(<ResetPasswordForm handleSubmit={mockHandler} />);

    const passwordInput = screen.getByLabelText(/Password/);
    const confirmPasswordInput = screen.getByLabelText(/Confirm your password/);

    await user.type(passwordInput, "Password1");
    await user.type(confirmPasswordInput, "Password1");

    const button = screen.getByTestId("resetPassword-button");
    await user.click(button);

    expect(passwordInput.value).toBe("Password1");
    expect(confirmPasswordInput.value).toBe("Password1");
    expect(mockHandler).toHaveBeenCalledTimes(1);
  });

  it("checks recover without email does not send email", async () => {
    const mockHandler = jest.fn();
    render(<ResetPasswordForm handleSubmit={mockHandler} />);

    const button = screen.queryByTestId("recover-button");
    await user.click(button);

    expect(mockHandler).toHaveBeenCalledTimes(0);
  });
});
