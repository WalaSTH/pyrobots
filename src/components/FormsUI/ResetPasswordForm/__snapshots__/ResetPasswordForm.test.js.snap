// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPasswordForm component rendering 1`] = `
<div
  className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-he3bwl-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h1
        className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Link Expired
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-189e9pj-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Back to login
      </button>
    </div>
  </div>
</div>
`;
